python/cudf/cudf/core/frame.py

```diff
@@ -182,16 +182,29 @@ def reassign_categories(categories, cols, col_idxs):
         # names with their integer positions in the `cols` list
         tables = []
         for cols in columns:
-            table_cols = cols[first_data_column_position:]
-            table_names = indices[first_data_column_position:]
-            table = cls(data=dict(zip(table_names, table_cols)))
+            table_index = None
             if 1 == first_data_column_position:
-                table._index = as_index(cols[0])
+                table_index = as_index(cols[0])
             elif first_data_column_position > 1:
-                index_cols = cols[:first_data_column_position]
-                index_names = indices[:first_data_column_position]
-                table._index = cls(data=dict(zip(index_names, index_cols)))
-            tables.append(table)
+                table_index = libcudf.table.Table(
```

Comment:
Are we able to just do the following?

```suggestion
                table_index = cls(
```

In retrospect, I should have used the name `_from_frame` instead of `_from_table` for constructing DataFrames/Series from frames :(

Reply:
I'm sure we can, but I figured we should use the constructor with the least amount of overhead.

Reply:
👍 Let's leave as is